src/platform_linux/platform.h

```diff
@@ -739,6 +756,18 @@ platform_enable_tracing_shm_frees();
  * Non-inline implementations belong in a .c file in the platform_* directory.
  * Declarations for the non-inline functions can go in platform_inline.h
  */
+/*
+ * Structure to encapsulate a {memory-addr, memory-size} pair. Used to track
+ * allocation and, more importantly, free of memory fragments for opaque
+ * "objects". Used typically to manage memory for arrays of things.
+ * The 'addr' field is intentionally -not- the 1st field, to reduce lazy
+ * programming which might try to bypass provided interfaces.
+ */
+typedef struct platform_memfrag {
+   size_t size;
+   void  *addr;
+} platform_memfrag;
```

Comment:
NOTE to myself: Move this to some private header and only expose `memfrag_init_size()` method, somehow.